cmd/senpai/main.go

```diff
@@ -51,16 +53,33 @@ func main() {
 	lastNetID, lastBuffer := getLastBuffer()
 	app.SwitchToBuffer(lastNetID, lastBuffer)
 
+	// Write last buffer on close
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM, syscall.SIGHUP)
+
+	go func() {
+		for sig := range sigCh {
+			switch sig {
+			case syscall.SIGHUP:
+				app.Close()
+				fmt.Println("SIGHUP")
+				lastBufferPath := getLastBufferPath()
+				lastNetID, lastBuffer = app.CurrentBuffer()
+				err = os.WriteFile(lastBufferPath, []byte(fmt.Sprintf("%s %s", lastNetID, lastBuffer)), 0666)
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "failed to write last buffer at %q: %s\n", lastBufferPath, err)
+				}
+			case syscall.SIGINT, syscall.SIGTERM:
+				app.Close()
+				fmt.Println("SIGTERM")
+				return
+			}
+		}
+	}()
+
 	app.Run()
 	app.Close()
-
-	// Write last buffer on close
-	lastBufferPath := getLastBufferPath()
-	lastNetID, lastBuffer = app.CurrentBuffer()
-	err = os.WriteFile(lastBufferPath, []byte(fmt.Sprintf("%s %s", lastNetID, lastBuffer)), 0666)
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "failed to write last buffer at %q: %s\n", lastBufferPath, err)
-	}
```

Comment:
You'll still want senpai to save the buffer when `app.Run` returns (user explicitly typed `/quit`).

Otherwise, PR looks good (provided app is reworked to allow `app.Close` be run concurrently of `app.Run`)